src/dialogs/create-file.tsx

```diff
@@ -0,0 +1,232 @@
+                      {_("Cancel")}
+                  </Button>
+              </>
+          }
+        >
+            <Stack>
+                {createError &&
+                <Alert
+                  className="file-editor-alert"
+                  variant="danger"
+                  title={createError}
+                  isInline
+                />}
+                <Form isHorizontal>
+                    <FormGroup
+                      label={_("File name")}
+                      fieldId="file-name"
+                    >
+                        <TextInput
+                          autoFocus // eslint-disable-line jsx-a11y/no-autofocus
+                          id="file-name"
+                          isRequired
+                          value={filename}
+                          onChange={(_event, value) => {
+                              setFileNameError(checkFilename(value, cwdInfo?.entries || {}, undefined));
+                              setFilename(value);
+                          }}
+                        />
+                        <FormHelper fieldId="file-name" helperTextInvalid={filenameError} />
+                    </FormGroup>
+                    <TextArea
+                      id="editor-text-area"
+                      className="file-editor"
+                      value={initialText}
+                      onChange={(_ev, content) => setInitialText(content)}
+                    />
+                    {candidates.length > 0 &&
+                        <FormGroup fieldId="create-file-owner" label={_("Owner")}>
+                            <FormSelect
+                              id="create-file-owner"
+                              value={owner}
+                              onChange={(_ev, val) => setOwner(val)}
+                            >
+                                {candidates.map(owner =>
+                                    <FormSelectOption
+                                      key={owner}
+                                      value={owner}
+                                      label={owner}
+                                    />)}
+                            </FormSelect>
+                        </FormGroup>}
+                </Form>
+            </Stack>
+        </Modal>
+    );
+};
+
+export async function show_create_file_dialog(
+    dialogs: Dialogs,
+    path: string,
+    addAlert: (title: string, variant: AlertVariant, key: string, detail?: string) => void
+) {
+    if (!superuser.allowed) {
+        try {
+            await cockpit.spawn(["test", "-w", path]);
+        } catch {
+            addAlert(_("Cannot create file in current directory"), AlertVariant.warning, "create-file",
+                     _("Permission denied"));
+            return;
+        }
+    }
+
+    // A cancelled dialog means no file is created so don't open the editor
+    try {
+        const filename = await dialogs.run(CreateFileModal, { path });
+        edit_file(dialogs, filename);
```

Comment:
As discussed on chat: The initial "Create file dialog":

![image](https://github.com/user-attachments/assets/3221c9a3-4494-456f-b099-517dbab8c0e1)

already has everything that we want to offer: name, owner, and file contents. This automatic immediate edit dialog

![image](https://github.com/user-attachments/assets/5d045dbb-fef4-4e54-81ff-b2e21af29662)

has nothing extra that the first dialog didn't already covered. I can even click "Close" there and the file is still created.

Is this meant for a future where the "edit" dialog becomes more powerful or so? But at least until then this extra call should go, it's rather confusing and unnecessary.